### PR TITLE
Add documentation for running GDB and LLDB

### PR DESCRIPTION
**Pull request by jonmeow**
After lots of fidgeting, LLDB seems to work. Not sure how reliable this will be though -- fission seems a little hit and miss.

### CODE REVIEW

**Comment by jonmeow** [comment]
After more digging, when building with --fission, it looks like _intermediate_ dwo files aren't provided on rebuilds. That is to say, things like `bazel-out/k8-dbg/bin/external/_main~llvm_project~llvm-project/llvm/_objs/Support/IntEqClasses.pic.dwo` aren't produced, even though they're mentioned by the dwarf. So yeah, this looks like a bazel issue with --fission.

File: docs/project/contribution_tools.md [hunk 1]
```diff
@@ -239,6 +240,44 @@ considering if they fit your workflow.
...
         -   **NOTE**: This assumes you have `python` 3 installed on your system.
 
+#### Using LLDB with VS Code
+
+The required setup for LLDB is:
+
+1.  Install a minimum of LLVM 19 instead of LLVM 16.
+    -   The `lldb-dap` tool was added as part of LLVM 19.
+2.  Symlink `lldb_launch.json` to `launch.json`:
+    `ln -s lldb_launch.json .vscode/launch.json`
```
Comment (chandlerc) [inline docs/project/contribution_tools.md]: Could we make it a bit more explicit what this is doing? Thinking about something like this:

```suggestion
2.  Symlink `lldb_launch.json` to `launch.json` in the workspace directory (typically the checkout of the `carbon-lang` repository):
    `ln -s lldb_launch.json .vscode/launch.json`
```

File: docs/project/contribution_tools.md [hunk 2]
```diff
@@ -247,9 +286,10 @@ comfortable with it. The essential CMake options to pass in order for this to
 work reliably include:
 
 ```
--DLLVM_ENABLE_PROJECTS=clang;clang-tools-extra;lld
+-DLLVM_ENABLE_PROJECTS=clang;clang-tools-extra;lld;lldbg
```
Comment (chandlerc) [inline docs/project/contribution_tools.md]:
```suggestion
-DLLVM_ENABLE_PROJECTS=clang;clang-tools-extra;lld;lldb
```

**Review by chandlerc — APPROVED** [review]
Niiice! Some tweaks inline, but LG with those.

File: docs/project/contribution_tools.md (hunk 2 above)
Comment (jonmeow) [reply docs/project/contribution_tools.md]: Done

File: docs/project/contribution_tools.md (hunk 1 above)
Comment (jonmeow) [reply docs/project/contribution_tools.md]: How about instead writing "In the `.vscode` subdirectory"?

**Pull request merged**